packages/atomic/src/components/insight/atomic-insight-interface/e2e/page-object.ts

```diff
@@ -68,42 +72,49 @@ export class InsightInterfacePageObject extends BasePageObject<'atomic-insight-i
     });
   }
 
-  get connectResultLink() {
-    return this.insightResults.getByRole('link', {name: 'Connect'});
+  get videoResultLink() {
+    return this.page.getByTestId('video-insight-result-link');
   }
 
-  async waitForNonVideoResultsToBeDetached() {
-    await this.connectResultLink.waitFor({state: 'detached'});
+  async waitForVideoResultLinksToBeVisible() {
+    await this.videoResultLink.first().waitFor({state: 'visible'});
+  }
+
+  async waitForInsightSearchResponse() {
+    await this.page.waitForResponse(INSIGHT_SEARCH_REGEX);
   }
 
   getResultByIndex(index: number) {
     return this.insightResults.nth(index);
   }
 
-  async hoverResultByIndex(index: number) {
-    await this.getResultByIndex(index).hover();
+  async hoverResultTitleByIndex(index: number) {
+    await this.getResultByIndex(index)
+      .locator('atomic-result-link')
+      .first()
+      .hover();
   }
 
   getActionBarByIndex(index: number) {
-    return this.getResultByIndex(index).locator(
-      'atomic-insight-result-action-bar'
-    );
+    return this.getResultByIndex(index)
+      .locator('atomic-insight-result-action-bar')
+      .first();
```

Comment:
Since there can be more than one result-action-bar per result because of folding, we pick the first which is the one for the parent result.